components/ws-proxy/pkg/proxy/routes.go

```diff
@@ -68,13 +70,18 @@ func NewRouteHandlerConfig(config *Config, opts ...RouteHandlerConfigOpt) (*Rout
 type RouteHandler = func(r *mux.Router, config *RouteHandlerConfig)
 
 // installWorkspaceRoutes configures routing of workspace and IDE requests.
-func installWorkspaceRoutes(r *mux.Router, config *RouteHandlerConfig, ip WorkspaceInfoProvider) {
+func installWorkspaceRoutes(r *mux.Router, config *RouteHandlerConfig, ip WorkspaceInfoProvider, hostKeyList []ssh.Signer) {
 	r.Use(logHandler)
 
 	// Note: the order of routes defines their priority.
 	//       Routes registered first have priority over those that come afterwards.
 	routes := newIDERoutes(config, ip)
 
+	// if host key is not empty, we use /__ws_proxy/ssh_host_key to provider public host key
+	if len(hostKeyList) > 0 {
+		routes.HandleSSHHostKeyRoute(r.Path("/__ws_proxy/ssh_host_key"), hostKeyList)
```

Comment:
@csweichel Do you mean to match path for `__ws_proxy` and have subroute for it?

Reply:
I meant not tying it to `ws-proxy`, e.g. calling it `__ssh` or something like that. Just so that if we ever move this functionality elsewhere, we're not left with a misnomer we can't move away from.

Reply:
@iQQBot Could you change it to something like `__ssh/host_keys`?

Reply:
Let's go with one :) I thought we are using two for supervisor, but it is one there as well.

Reply:
@csweichel Could you have a look again please? 🙏